core/vnl/vnl_matrix.h

```diff
@@ -300,23 +282,114 @@ class VNL_EXPORT vnl_matrix
 
 
   //: Add rhs to each element of lhs matrix and return result in new matrix
-  vnl_matrix<T> operator+(T const& v) const { return vnl_matrix<T>(*this, v, vnl_tag_add()); }
+  vnl_matrix<T> operator+(T const& v) const {
+    vnl_matrix<T> result(this->rows(), this->cols());
+    const unsigned int n = this->num_rows * this->num_cols;
+    T const *m = this->data[0];
+    T *dst = result.data[0];
+
+    for (unsigned int i = 0; i < n; ++i)
+      dst[i] = T(m[i] + v);
+    return result;
```

Comment:
`return result` is potentially expensive, because `vnl_matrix` does not have a fast move-constructor. I think it would be better to postpone this change until we have fast move-semantics for `vnl_matrix`. (Because I think you cannot fully rely on NRVO -- named return value optimization.)

Reply:
done.